src/thorin/be/ll/ll.cpp

```diff
@@ -330,6 +352,14 @@ void CodeGen::emit_epilogue(Lam* lam) {
                 args.emplace_back(convert(arg->type()) + " " + emitted_arg);
         }
 
+        if (app->args().back()->isa<Bot>()) {
+            // TODO: Perhaps it'd be better to simply η-wrap this prior to the BE...
+            assert(convert_ret_pi(app->callee_type()->ret_pi()) == "void");
+            bb.tail("call void {}({, })", emmited_callee, args);
+            return bb.tail("unreachable");
+        }
```

Comment:
Yes. I think it would be nice to extend RetWap to ensure that all branches and return-continuations are `Lam`s.
I think that's essentially on what's happening on master right now, since after PE only `Lam`s and return-continuations should be left.